### ISSUE TITLE
Config.xml is reset to previous version in case of power failure

### ISSUE THREAD

**Issue by 1pratham** [floft/BellSystem#2]
I make some changes in the schedule and other things save all my bell system settings. Let's say this as setting1. I shutdown/restart Pi properly. All the settings are saved and restored properly. Now I make some changes in the settings and save them. Let's say this as setting2. They work properly as long as Pi is working. However, in case there is a power failure and Pi shuts down abruptly. On starting Pi again, all the setting2 is lost and I see all the schedules and everything as per last properly saved setting i.e setting1. 
All the other things are working fine. I really appreciate the work you have put in and shared it with others. Thanks!

**Comment by floft** [comment]
Yeah, I'm not too surprised. This means it's probably not syncing to disk between the save and the power failure. This depends on the filesystem, filesystem options, etc. It also depends on how long a time period that is. I've tested power failure when using Arch Linux ARM, but I'm not sure how long I waited before killing it. Evidently it was long enough for it to sync to disk, so I didn't notice the problem.

We probably just need to call "sync" after saving the file. I found some sort of fsync in PHP. I'll look into that. (Alternatively, I could store the data in something like SQLite, which would be better for the website and crash resistance but harder to then edit via the terminal.)

**Comment by 1pratham** [comment]
Is some other file also involved apart from config.xml? 
I have done the following test:

1) Make changes using the WebUI.
2) Manually check changes are reflected in config.xml.
3) Plug out the power source after 3-4 minutes.
4) Start the Pi again.

Even though changes are reflected in Step 2, on starting the Pi in Step 4 the older settings are again reflected in the WebUI and config.xml. 

As a temporary solution, I have added a "Restart" button in the WebUI which safely restarts the Pi. So after each proper setting save, I just press the restart button.

**Comment by floft** [comment]
The website interface more-or-less only changes the config.xml file (though, probably some session files for login too I guess). A forced sync to disk on save should fix the problem. I added that in c69589493d4a6970c499f8a477a23235fdeb661e. See if this works. I pulled the power after a few seconds after clicking "save" and all the changes stayed intact. Though, I also tried it without the "system('sync')" in the PHP code and I didn't lose any data for some reason. Maybe there's a difference in SD card write speeds between your and my cards.

Interestingly, there has been some discussion about handling proper shutdown on power failure: https://raspberrypi.stackexchange.com/questions/5372/circuit-to-safely-power-down-pi

For example, one of the UPS / batteries that was listed states "Full power management API available to Raspberry Pi OS with auto shutdown capability when running low on batteries": https://uk.pi-supply.com/products/pijuice-standard

That would be the ideal solution in general, which would keep it online during brief power failures and then allow proper shutdown when the battery is getting low. At least with my setup of bell systems though, the bell won't work when the power is out, so the Pi dying is perfectly fine as long as the filesystem doesn't get corrupted.

Let me know if the update works for you.
